LargeBarrelAnalysis/Downscaler.cpp

```diff
@@ -0,0 +1,98 @@
+/**
+ *  @copyright Copyright 2022 The J-PET Framework Authors. All rights reserved.
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may find a copy of the License in the LICENCE file.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *  @file Downscaler.cpp
+ */
+
+#include "Downscaler.h"
+#include <JPetOptionsTools/JPetOptionsTools.h>
+#include <iostream>
+
+using namespace jpet_options_tools;
+using namespace std;
+
+Downscaler::Downscaler(const char* name) : JPetUserTask(name)
+{
+  std::random_device rd;
+  fRandomGenerator = std::mt19937(rd());
+}
+
+Downscaler::~Downscaler() {}
+
+bool Downscaler::init()
+{
+  INFO("Event downscaling started.");
+
+  // Parameter for back to back categorization
+  if (isOptionSet(fParams.getOptions(), fDownscalingRatesKey))
+  {
+    fDownscalingRates = getOptionAsVectorOfDoubles(fParams.getOptions(), fDownscalingRatesKey);
```

Comment:
```suggestion
    fDownscalingRates = getOptionAsVectorOfDoubles(fParams.getOptions(), fDownscalingRatesKey);
    std::transform(fDownscalingRates.begin(), fDownscalingRates.end(), fDownscalingRates.begin(), [](double el)->double { return el/100.;});
```
Since later you always use values divided by 100, maybe it would be better to divide it already here?